runtime/jcl/common/java_lang_invoke_MethodHandleNatives.cpp

```diff
@@ -1089,14 +1165,16 @@ Java_java_lang_invoke_MethodHandleNatives_resolve(
 			if ((0 != target) && ((0 != vmindex) || J9_ARE_ANY_BITS_SET(flags, MN_IS_METHOD | MN_IS_CONSTRUCTOR))) {
 				/* Refetch reference after GC point */
 				membernameObject = J9_JNI_UNWRAP_REFERENCE(self);
-				J9VMJAVALANGINVOKEMEMBERNAME_SET_FLAGS(currentThread, membernameObject, new_flags);
-				J9VMJAVALANGINVOKEMEMBERNAME_SET_CLAZZ(currentThread, membernameObject, new_clazz);
-				J9OBJECT_U64_STORE(currentThread, membernameObject, vm->vmindexOffset, (U_64)vmindex);
-				J9OBJECT_U64_STORE(currentThread, membernameObject, vm->vmtargetOffset, (U_64)target);
+				if (addMemberNameToClass(currentThread, membernameObject, new_clazz)) {
```

Comment:
We are adding to the list from three different locations: init, resolve and expand. @fengxue-IS Can you confirm is the answer to the below questions is correct?
- Will we catch all MemberNames that are created from these three locations? **Yes**
- Can duplicate entries be added to the list? **No**

Reply:
Maybe my description of "resolved" `MemberName`s was too loose. What we really need - and what this code adds to the list - are those `MemberName` instances that have `vmtarget`/`vmindex` set based on a particular member of some class. (It's `vmtarget` especially that's important, though `vmindex` also matters when using the extensions.) From grepping, these are the only two locations I found where these fields are stored to (except for `prepareToFixMemberNames()` and `fixMemberNames()`, which temporarily repurpose them, and then only in objects where they've already been set.) It's my impression that this corresponds more or less to what the Java source considers to be a resolved `MemberName`, but I haven't checked that it corresponds exactly

Reply:
There is an other place that sets the MemberName.clazz field in `Java_java_lang_invoke_MethodHandleNatives_expand`. Though that path doesn't really modify the dependency. For now I think its safe to assume all resolved MN should be caught by these two locations

Reply:
There is an other place that sets the MemberName.clazz field in Java_java_lang_invoke_MethodHandleNatives_expand. Though that path doesn't really modify the dependency. For now I think its safe to assume all resolved MN should be caught by these two locations

Reply:
If `clazz` is set without `vmtarget`/`vmindex`, then no fixup is necessary